eis_toolkit/raster_processing/unique_combinations.py

```diff
@@ -0,0 +1,75 @@
+from typing import Dict, List, Tuple
+
+import numpy as np
+import rasterio
+
+from eis_toolkit.exceptions import InvalidParameterValueException
+
+
+def _unique_combinations(
+    bands: List[np.ndarray],
+) -> np.ndarray:
+
+    unique_combinations: Dict[int, int] = {}
+    unique_indices = np.zeros(bands[0].shape, dtype=int)
+    combination = 1
+
+    for band in bands:
+        for row in range(band.shape[0]):
+            for column in range(band.shape[1]):
+                raster_value = band[row, column]
+                if raster_value not in unique_combinations:
+                    unique_combinations[raster_value] = combination
+                    combination += 1
+                unique_indices[row, column] = unique_combinations[raster_value]
+
+    return unique_indices
+
+
```

Comment:
Please add @beartype decorator here. We picked up this handy package to do input validation some time ago, you can check some existing functions for reference.